src/coreComponents/dataRepository/wrapperHelpers.hpp

```diff
@@ -515,11 +515,32 @@ pushDataToConduitNode( ArrayOfArrays< T, INDEX_TYPE > const & var2,
   conduit::DataType const sizesType( conduitTypeInfo< INDEX_TYPE >::id, numArrays );
   node[ "__sizes__" ].set_external( sizesType, const_cast< void * >( static_cast< void const * >( sizes ) ) );
 
-  // Push the data into conduit
+  // **** WARNING: alters the uninitialized values in the ArrayOfArrays ****
+  T const * const values = var.getValues();
+  std::vector< T > valuesCopy( values, values + offsets[numArrays] );
+  for( INDEX_TYPE i = 0; i < numArrays; ++i )
+  {
+    INDEX_TYPE const curOffset = offsets[ i ];
+    INDEX_TYPE const nextOffset = offsets[ i + 1 ];
+    for( INDEX_TYPE j = curOffset + var.sizeOfArray( i ); j < nextOffset; ++j )
+    {
+      if constexpr ( std::is_arithmetic< T >::value )
```

Comment:
Isn't this redundant since `T() = 0` if `std::is_arithmetic< T >::value == true`? Also, why do you choose to make a copy instead of changing the values in the array? If we're going to make a copy anyways I'd suggest compressing out the uninitialized values. The only "problem" with this is that when we read the array back in it is compressed as well.

Reply:
> Isn't this redundant since `T() = 0` if `std::is_arithmetic< T >::value == true`? Also, why do you choose to make a copy instead of changing the values in the array? If we're going to make a copy anyways I'd suggest compressing out the uninitialized values. The only "problem" with this is that when we read the array back in it is compressed as well.

- I wasn't sure that `T() = 0` was actually true. If we have a variable declared as `int a;`, is `a` guaranteed to be equal to zero?
- I choose to make a copy because I didn't want to break the `const` promise on the write function. It is of course a double edge sword, because you are not reading in what was passed to the write function...but then changing the `const` object that was passed to the write function seems somewhat illegal? Does that make sense?
- I did not want to compress it because I wanted to make sure that the behavior of a restart run is identical to the original run. Compressing the `ArrayOfArrays` would result in behavioral changes....which of course should be OK...unless there is some sort of allocation error that occurs when you try to insert new values in the `ArrayOfArrays`...etc. This is of course ignoring the fact that the array you read back in will not be exactly identical to the one you wrote out in the "unallocated" values....but those shouldn't matter ever or we have bigger problems.

Reply:
- `int a;` is different from `int a = int()`. The first is uninitialized and can be anything, the second calls the default constructor which for numeric types always returns zero.
- Yeah I think breaking correctness is fine here.
- I get that, if you want as similar behavior as possible then setting the uninitialized values is definitely the way to go.

Reply:
@corbett5  
I should know that. thanks!